src/common/kyc/index.js

```diff
@@ -211,39 +211,76 @@ const loadRelyingPartyOperation = rpName => async (dispatch, getState) => {
 	}
 	const config = rp.relying_party_config;
 
-	try {
-		const session = mpService.createRelyingPartySession(config);
-		await session.establish();
-		let templates = await Promise.all(
-			(await session.listKYCTemplates()).map(async tpl => {
-				const id = tpl.id || tpl.templateId;
-				tpl = await session.getKYCTemplate(id);
-				tpl.id = id;
-				return tpl;
-			})
-		);
-
-		const applications = await session.listKYCApplications();
-		await dispatch(
-			kycActions.updateRelyingParty({
-				name: rpName,
-				description: rp.description,
-				templates,
-				applications,
-				session,
-				lastUpdated: ts
-			})
-		);
-	} catch (error) {
-		await dispatch(
-			kycActions.updateRelyingParty(
-				{
+	if (!authenticate) {
+		try {
+			const session = mpService.createRelyingPartySession(config);
+			let templates = await Promise.all(
+				(await session.listKYCTemplates()).map(async tpl => {
+					const id = tpl.id || tpl.templateId;
+					tpl = await session.getKYCTemplate(id);
+					tpl.id = id;
+					return tpl;
+				})
+			);
+
+			const applications = [];
+
+			await dispatch(
+				kycActions.updateRelyingParty({
 					name: rpName,
+					description: rp.description,
+					templates,
+					applications,
+					session,
 					lastUpdated: ts
-				},
-				error
-			)
-		);
+				})
+			);
+		} catch (error) {
+			await dispatch(
+				kycActions.updateRelyingParty(
+					{
+						name: rpName,
+						lastUpdated: ts
+					},
+					error
+				)
+			);
+		}
+	} else {
+		try {
+			const session = mpService.createRelyingPartySession(config);
```

Comment:
@andregoncalves I think a better way is to move this to a helper function like await getSession(config, authenticate). Cheers

Reply:
Yep, good idea.